internal/controller/package_controller.go

```diff
@@ -544,3 +557,58 @@ func (r *PackageReconcilationContext) pruneOwnedPackageInfos(ctx context.Context
 	}
 	return compositeErr
 }
+
+func (r *PackageReconcilationContext) pruneOwnedPackages(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+	var errs error
+	ownedPackagesCopy := r.pkg.Status.OwnedPackages[:]
+OuterLoop:
+	for _, ref := range r.pkg.Status.OwnedPackages {
+		for _, newRef := range r.currentOwnedPackages {
+			if ownerutils.RefersToSameResource(ref, newRef) {
+				continue OuterLoop
+			}
+		}
+
+		r.setShouldUpdate(ownerutils.Remove(&ownedPackagesCopy, ref))
```

Comment:
I think it would be more resilient to only remove the reference if we know that either:
- it doesn't exist anymore (get returns 404 error)
- it doesn't have an owner reference to this pkg anyways
- the update removing the owner reference has succeeded

In the edge case of the update succeeding but the deletion after removing the last owner reference failing, I think it would be OK for now that the package remains as an orphan.